Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG NODE_VERSION="12.2.0"
+FROM library/node:${NODE_VERSION} AS development
+
+ENV NODE_ENV "development"
+WORKDIR /usr/src/freshlytics
+
+COPY package*.json ./
+RUN npm install --no-optional
+
+COPY . .
+
+CMD ["npm", "run", "dev"]
+EXPOSE 3000/tcp 3001/tcp
+
+FROM development AS builder
+
+ENV NODE_ENV "production"
+RUN npm run build
+
+FROM library/node:${NODE_VERSION}-slim AS production
+
+ENV NODE_ENV "production"
+WORKDIR /usr/src/freshlytics
+
+ENV BUILD_DEPS="python2.7 make g++"
+
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+    ${BUILD_DEPS} && \
+  rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /usr/src/freshlytics/package*.json ./
+RUN PYTHON=/usr/bin/python2.7 \
```

Comment:
Is it still necessary to run `npm install` after we've run `npm run build` in the previous stage?

Reply:
tl;dr: during build step you have typescript (& friends) installed to transpile code. production step has runtime dependencies 😊

The reason you would need to run `npm install` during the production stage, is because you will need the dependencies installed for runtime, and you wouldn't want to copy over the node_modules from the previous stages, as those include the devDependencies, which have typescript, etc that are necessary for transpiling ts files.

Reply:
Thanks, that makes sense! 👍